src/openzeppelin/token/erc1155/library.cairo

```diff
@@ -0,0 +1,783 @@
+        )
+        return ()
+    end
+
+    func _burn{
+            syscall_ptr: felt*,
+            pedersen_ptr: HashBuiltin*,
+            range_check_ptr
+        }(from_: felt, id: Uint256, amount: Uint256):
+        alloc_locals
+        with_attr error_message("ERC1155: burn from the zero address"):
+            assert_not_zero(from_)
+        end
+
+        # Check uints valid
+        with_attr error_message("ERC1155: id is not a valid Uint256"):
+            uint256_check(id)
+        end
+        with_attr error_message("ERC1155: amount is not a valid Uint256"):
+            uint256_check(amount)
+        end
+
+        # Deduct from burner
+        let (from_balance: Uint256) = ERC1155_balances.read(id, from_)
+        with_attr error_message("ERC1155: burn amount exceeds balance"):
+            let (new_balance: Uint256) = SafeUint256.sub_le(from_balance, amount)
+        end
+        ERC1155_balances.write(id, from_, new_balance)
+
+        let (operator) = get_caller_address()
+        TransferSingle.emit(operator=operator, from_=from_, to=0, id=id, value=amount)
+        return ()
+    end
+
+    func _burn_batch{
+            syscall_ptr: felt*,
+            pedersen_ptr: HashBuiltin*,
+            range_check_ptr
+        }(
+            from_: felt,
+            ids_len: felt,
+            ids: Uint256*,
+            amounts_len: felt,
+            amounts: Uint256*
+        ):
+        alloc_locals
+        with_attr error_message("ERC1155: burn from the zero address"):
+            assert_not_zero(from_)
+        end
+        with_attr error_message("ERC1155: ids and amounts length mismatch"):
+            assert ids_len = amounts_len
+        end
+
+        # Recursive call
+        let len = ids_len
+        burn_batch_iter(from_, len, ids, amounts)
+        let (operator) = get_caller_address()
+        TransferBatch.emit(
+            operator=operator,
+            from_=from_,
+            to=0,
+            ids_len=ids_len,
+            ids=ids,
+            values_len=amounts_len,
+            values=amounts
+        )
+        return ()
+    end
+
+    func _set_approval_for_all{
+            syscall_ptr: felt*,
+            pedersen_ptr: HashBuiltin*,
+            range_check_ptr
+        }(owner: felt, operator: felt, approved: felt):
+        # check approved is bool
+        assert approved * (approved - 1) = 0
+        # caller/owner already checked  non-0
+        with_attr error_message("ERC1155: setting approval status for zero address"):
+            assert_not_zero(operator)
+        end
+        with_attr error_message("ERC1155: setting approval status for self"):
+            assert_not_equal(owner, operator)
+        end
+        ERC1155_operator_approvals.write(owner, operator, approved)
+        ApprovalForAll.emit(owner, operator, approved)
+        return ()
+    end
+
+    func _set_uri{
+            syscall_ptr: felt*,
+            pedersen_ptr: HashBuiltin*,
+            range_check_ptr
```

Comment:
I think we should expose this in ERC1155_Mintable_Burnable.cairo so we can test it. Since it's a preset, I'd include the `assert_only_owner`